### PR TITLE
Flax schedulers: replace control flow with functions

### PR DESCRIPTION
**Pull request by pcuenca**
Otherwise jitting/parallelization don't work properly as they don't know how to deal with traced objects.

I started with `FlaxPNDMScheduler`, and temporarily removed `step_prk`.

### CODE REVIEW

**Comment by pcuenca** [comment]
We can test with just `pjit` like this:

```Python
import jax
import jax.numpy as jnp

from diffusers import FlaxPNDMScheduler

scheduler = FlaxPNDMScheduler.from_config(PATH_TO_SCHEDULER_DIR)

latents_shape = (1, 64, 64, 3)
scheduler_state = scheduler.set_timesteps(
    scheduler.state,
    shape = latents_shape,         # Needs to be known in advance to reserve space
    num_inference_steps = 50,
)

key1, key2 = jax.random.split(jax.random.PRNGKey(0))
latents = jax.random.normal(key1, shape=latents_shape, dtype=jnp.float32)
noise = jax.random.normal(key2, shape=latents_shape, dtype=jnp.float32)

p_step = jax.jit(scheduler.step, static_argnums=4)
latents, scheduler_state = p_step(scheduler_state, noise, 37, latents, return_dict=False)
```

This example should work with both `pjit` and without (invoking `scheduler.step` directly), for all schedulers.
I'll prepare a more complete example with `pmap` later.

**Comment by patrickvonplaten** [comment]
Cool! Will focus on DDIM for now to get the pipeline working with it

**Comment by pcuenca** [comment]
Replaced by #583 for PNDM. We'll open separate PRs for others.
